x/stake/endblock.go

```diff
@@ -12,8 +12,17 @@ import (
 
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.ValidatorUpdate, completedUbds []types.UnbondingDelegation) {
 	// only change validator set in breath block after BEP159
-	var events sdk.Events
-	var csEvents sdk.Events
+	var (
+		events             sdk.Events
+		csEvents           sdk.Events
+		refundEvents       sdk.Events
+		closeChannelEvents sdk.Events
+	)
+	if sdk.IsUpgrade(sdk.BCFusionSecondHardFork) {
+		refundEvents = handleRefundCrossStake(ctx, k)
```

Comment:
We need undelegated all the staking, dont we?

Reply:
yes, I change the handleRefundCrossStake function, please check